vms/platformvm/block/executor/verifier.go

```diff
@@ -394,43 +395,67 @@ func (v *verifier) standardBlock(
 	b *block.ApricotStandardBlock,
 	onAcceptState state.Diff,
 ) error {
-	blkState := &blockState{
+	inputs, atomicRequests, onAcceptFunc, err := v.processStandardTxs(b.Transactions, onAcceptState, b.Parent())
+	if err != nil {
+		return err
+	}
+
+	blkID := b.ID()
+	v.blkIDToState[blkID] = &blockState{
 		statelessBlock: b,
-		onAcceptState:  onAcceptState,
+
+		onAcceptState: onAcceptState,
+		onAcceptFunc:  onAcceptFunc,
+
 		timestamp:      onAcceptState.GetTimestamp(),
-		atomicRequests: make(map[ids.ID]*atomic.Requests),
+		inputs:         inputs,
+		atomicRequests: atomicRequests,
 	}
 
-	// Finally we process the transactions
-	funcs := make([]func(), 0, len(b.Transactions))
-	for _, tx := range b.Transactions {
+	v.Mempool.Remove(b.Transactions)
```

Comment:
nit: I know this doesn't matter when everything is behind a single lock, but it feels better to me to remove transactions from the mempool before we add to the `blkIDToState` map (which is when other processes could conceivably query this info, assuming that all related txs are no longer in the mempool).

Reply:
done